### PR TITLE
change tests to correctly check the first digits of area/exchange codes

### PR DESCRIPTION
**Pull request by LyleCharlesScott**
* Fixed test cases of the form 123-456-7890, because those should fail as per the README (better matches canonical test cases). Additionally changed the numbers on other test cases to test only one factor at a time by not also breaking the area/exchange code rules.
* Removed 'invalid with right number of digits but letters mixed in' because of decision to remove it from the canonical tests. 
* Added missing 'valid when 11 digits and starting with 1 even with punctuation' from canonical tests.
* Added wording '... does not start with a 1' to 'invalid when 11 digits' as per canonical tests.
* Added area/exchange code testing from canonical tests.
* Added test cases to check whether 11-digit numbers starting with 1, but with incorrect area/exchange codes will fail appropriately (e.g., 1-123-456-7890, 1-223-045-7890 should fail).

Note that the example.js file needs an update. It does not test for area/exchange codes like the older js example.js test did.

### CODE REVIEW

File: exercises/phone-number/phone-number.spec.js [hunk 1]
```diff
@@ -2,52 +2,78 @@ import PhoneNumber from './phone-number';
...
+  });
+
+  xtest('invalid if exchange code starts with 0 or 1', () => {
+    const phone1 = new PhoneNumber('(223) 056-7890');
+    const phone2 = new PhoneNumber('(223) 156-7890');
+    expect(phone1.number()).toEqual(null);
+    expect(phone2.number()).toEqual(null);
+  });
+
+  xtest('invalid when 11 digits starting with 1, ' +
```
Comment (tejasbubane) [inline exercises/phone-number/phone-number.spec.js]: I see that this set of test cases is not in canonical. This check is covered under two tests above which check for correct area code and exchange code. I think these are not required.

Comment (LyleCharlesScott) [reply exercises/phone-number/phone-number.spec.js]: The following code passes all of the previous test cases. It produces invalid output such as "0234567890" and "2231567890" in these edge cases because it accepts any 11 digit number starting with 1 as valid. None of the other tests cover this possibility.

export default class PhoneNumber {

  constructor(number) {
    this.rawNumber = number;
  }

  number() {
    if (/[a-zA-Z]/.test(this.rawNumber)) {
      return null;
    }

    return this._cleanedNumber();
  }

  _cleanedNumber() {
    const num = this.rawNumber.replace(/\D/g, '');

    if (num.length === 10 && num[0] >= 2 && num[3] >= 2) {
      return num;
    }

    if (num.length === 11 && num[0] === '1') {
      return num.substr(1);
    }

    return null;
  }
}
